lm/vocabulary.py

```diff
@@ -195,3 +196,38 @@ def run(self):
                 vocabulary_with_counts.write(f"{word} {count}\n")
 
         self.out_counter.set(counter)
+
+
+class GetOovRateJob(Job):
+    """
+    Obtains the OOV rate of a text file given a vocabulary file with one word per line.
+    """
+
+    def __init__(self, text_file: tk.Path, vocab_file: tk.Path):
+        self.text_file = text_file
+        self.vocab_file = vocab_file
+
+        self.out_oov_rate = self.output_var("oov_rate")
+
+        self.rqmt = {"cpu": 1, "mem": 5.0, "time": 2.0}
+
+    def tasks(self):
+        yield Task("run", resume="run", rqmt=self.rqmt)
```

Comment:
Oh, I do feel like this should be a `mini_task`?

Reply:
There are potentially many words to be loaded in the vocabulary (I'm not sure 5 GB are enough for 250-500k vocab size), and the job would take a relatively long time with any medium/large-sized text file. I interpret a mini task to be a task with minimal computation, like writing a small script to execute later.

Because of this, I'd tend to leave this as a cluster task and not as a mini task. If you still have issues with this, let me know and we can discuss this further.

Reply:
That makes sense, for some reason my gut feeling was that it would have been much higher. I'll lower the amount to 1-2 GB. I still feel like it's not suitable to be a mini task, even with the small memory footprint.

Reply:
Have you tried out running one of these tasks w/ a 500k vocab?